src/main/java/seedu/edrecord/model/group/Group.java

```diff
@@ -0,0 +1,70 @@
+package seedu.edrecord.model.group;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.edrecord.model.module.ModuleSystem;
+
+/**
+ * Represents a student's class group in EdRecord.
+ * Guarantees: immutable; is always valid
+ */
+public class Group {
+
+    public static final String MESSAGE_CONSTRAINTS = "Class code cannot have whitespaces.";
+    public static final String MESSAGE_DOES_NOT_EXIST = "Class with that code has yet to be created.";
+    public static final String MESSAGE_DUPLICATE = "Class with that code has already been created.";
+
+    /*
+     * The class code must not have any whitespace characters.
+     */
+    public static final String VALIDATION_REGEX = "[^\\s]+";
+
+    public static final ModuleSystem GROUP_SYSTEM = new ModuleSystem();
```

Comment:
Is this supposed to be of type `GroupSystem`?

Reply:
Nope, not supposed to be there. Updated